gapic/templates/%namespace/%name_%version/%sub/services/%service/_shared_macros.j2

```diff
@@ -139,26 +139,39 @@ def _get_http_options():
     return response
 {% endmacro %}
 
-{% macro rest_call_method_common(body_spec, method_name, service_name) %}
+{% macro rest_call_method_common(body_spec, method_name, service_name, is_async=False) %}
+{% set await_prefix = "await " if is_async else "" %}
+{% set async_prefix = "async " if is_async else "" %}
+{% set async_class_prefix = "Async" if is_async else "" %}
 
     http_options = _Base{{ service_name }}RestTransport._Base{{method_name}}._get_http_options()
+    {% if not is_async %}
```

Comment:
Add a todo comment with a link to a bug if this is a feature gap that needs to be fixed later

Reply:
Addressed.